examples/fabric-bridge-app/fabric-bridge-common/src/ZCLCallbacks.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DeviceManager.h"
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <lib/support/ZclString.h>
+
+using namespace ::chip;
+using namespace ::chip::app::Clusters;
+
+#define ZCL_DESCRIPTOR_CLUSTER_REVISION (1u)
+#define ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_CLUSTER_REVISION (2u)
+#define ZCL_BRIDGED_DEVICE_BASIC_INFORMATION_FEATURE_MAP (0u)
+
+// External attribute read callback function
+Protocols::InteractionModel::Status emberAfExternalAttributeReadCallback(EndpointId endpoint, ClusterId clusterId,
+                                                                         const EmberAfAttributeMetadata * attributeMetadata,
+                                                                         uint8_t * buffer, uint16_t maxReadLength)
+{
+    uint16_t endpointIndex  = emberAfGetDynamicIndexFromEndpoint(endpoint);
+    AttributeId attributeId = attributeMetadata->attributeId;
+
+    Device * dev = DeviceMgr().GetDevice(endpointIndex);
+    if (dev != nullptr && clusterId == app::Clusters::BridgedDeviceBasicInformation::Id)
+    {
+        using namespace app::Clusters::BridgedDeviceBasicInformation::Attributes;
+        ChipLogProgress(NotSpecified, "HandleReadBridgedDeviceBasicAttribute: attrId=%d, maxReadLength=%d", attributeId,
+                        maxReadLength);
+
+        if ((attributeId == Reachable::Id) && (maxReadLength == 1))
+        {
+            *buffer = dev->IsReachable() ? 1 : 0;
+        }
+        else if ((attributeId == NodeLabel::Id) && (maxReadLength == 32))
```

Comment:
Why is this `maxReadLength == 1` and not `maxReadLength >= 1`

Reply:
because 'Reachable' type is bool, it must be always 1 byte 

        <attribute side="server" code="17" define="REACHABLE"                 type="boolean"